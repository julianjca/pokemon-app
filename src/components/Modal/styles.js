import styled from "@emotion/styled";

export const ModalWindow = styled.div`
  position: fixed;
  background-color: rgba(0, 0, 0, 0.4);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
`;

export const Card = styled.div`
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),
    0 4px 6px -2px rgba(0, 0, 0, 0.05);
  background: ${(props) => props.theme.colors.background};
  padding: 30px 20px;
  box-sizing: border-box;
  border-radius: 10px;
  width: 95%;
  max-width: 350px;

  @media (min-width: ${(props) => props.theme.breakpoints.md}) {
    max-width: 500px;
  }
`;
