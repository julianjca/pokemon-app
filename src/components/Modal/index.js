import React from "react";
import { Card, ModalWindow } from "./styles";

const Modal = ({ children, handleCloseModal }) => {
  return (
    <ModalWindow onClick={handleCloseModal}>
      <Card data-testid="modal-card" onClick={(e) => e.stopPropagation()}>
        {children}
      </Card>
    </ModalWindow>
  );
};

export default Modal;
