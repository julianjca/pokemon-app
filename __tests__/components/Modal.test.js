/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { matchers } from "@emotion/jest";

import { render, fireEvent } from "../../test-utils";
import Modal from "../../src/components/Modal";

expect.extend(matchers);

describe("Input", () => {
  test("renders the component", () => {
    const element = render(<Modal />);

    expect(element).toMatchSnapshot();
  });

  test("renders children", () => {
    const element = render(
      <Modal>
        <div>children element</div>
      </Modal>
    );

    expect(element.getByText("children element")).toBeInTheDocument();
  });

  test("not running handleCloseModal when the children is clicked", () => {
    const handleCloseModal = jest.fn();
    const element = render(
      <Modal handleCloseModal={handleCloseModal}>
        <div>children element</div>
      </Modal>
    );

    const modalCard = element.getByTestId("modal-card");
    fireEvent.click(modalCard);

    expect(handleCloseModal).not.toHaveBeenCalled();
  });
});
