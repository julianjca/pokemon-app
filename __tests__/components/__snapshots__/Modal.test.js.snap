// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  position: fixed;
  background-color: rgba(0, 0, 0, 0.4);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

.emotion-2 {
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
  background: #ffffff;
  padding: 30px 20px;
  box-sizing: border-box;
  border-radius: 10px;
  width: 95%;
  max-width: 350px;
}

@media (min-width: 720px) {
  .emotion-2 {
    max-width: 500px;
  }
}

<body>
    <div>
      <div
        class="emotion-0 emotion-1"
      >
        <div
          class="emotion-2 emotion-3"
          data-testid="modal-card"
        />
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  position: fixed;
  background-color: rgba(0, 0, 0, 0.4);
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 999;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
}

.emotion-2 {
  box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1),0 4px 6px -2px rgba(0, 0, 0, 0.05);
  background: #ffffff;
  padding: 30px 20px;
  box-sizing: border-box;
  border-radius: 10px;
  width: 95%;
  max-width: 350px;
}

@media (min-width: 720px) {
  .emotion-2 {
    max-width: 500px;
  }
}

<div>
    <div
      class="emotion-0 emotion-1"
    >
      <div
        class="emotion-2 emotion-3"
        data-testid="modal-card"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
